packages/x-data-grid/src/components/panel/filterPanel/GridFilterInputValue.tsx

```diff
@@ -20,6 +20,16 @@ export type GridTypeFilterInputValueProps = GridFilterInputValueProps &
 
 type ItemPlusTag = GridFilterItem & { fromInput?: string };
 
+const sanitizeFilterItemValue = (value: any, type: GridTypeFilterInputValueProps['type']) => {
```

Comment:
Hi @arminmeh , I liked the approach we have used in https://github.com/mui/mui-x/pull/15252, using the same approach here as well.

Reply:
lint: `function sanitize...` and place it after the component